### ISSUE TITLE
Crash when trying to give unit in city center to liberated city-state

### ISSUE THREAD

**Issue by XardasXP** [yairm210/Unciv#10011]
**Platform**
Ubuntu, 23.04, 64bit.

**Version**
4.7.17-patch1 from Flathub

**Describe the bug**
Crash when trying to give unit stationed in liberated city-state city center to that city-state.

**To Reproduce**
1. Liberate City-State
2. Click on your unit inside City-State city center.
3. Click "Show more" in unit menu.
4. Error.

```
java.lang.Exception: Unit Destroyer at Tile @(25.0,17.0), Tyre, Tundra, Deer, City center can't be put in tile (25.0,17.0)!
  at com.unciv.logic.map.mapunit.MapUnit.putInTile(MapUnit.kt:649)
  at com.unciv.logic.map.mapunit.UnitMovement.canUnitSwapToReachableTile(UnitMovement.kt:471)
  at com.unciv.logic.map.mapunit.UnitMovement.access$canUnitSwapToReachableTile(UnitMovement.kt:15)
  at com.unciv.logic.map.mapunit.UnitMovement$getUnitSwappableTiles$1.invoke(UnitMovement.kt:438)
  at com.unciv.logic.map.mapunit.UnitMovement$getUnitSwappableTiles$1.invoke(UnitMovement.kt:438)
  at kotlin.sequences.FilteringSequence$iterator$1.calcNext(Sequences.kt:171)
  at kotlin.sequences.FilteringSequence$iterator$1.hasNext(Sequences.kt:194)
  at kotlin.sequences.SequencesKt___SequencesKt.none(_Sequences.kt:2130)
  at com.unciv.ui.screens.worldscreen.unit.actions.UnitActions.addSwapAction(UnitActions.kt:117)
  at com.unciv.ui.screens.worldscreen.unit.actions.UnitActions.getAdditionalActions(UnitActions.kt:94)
  at com.unciv.ui.screens.worldscreen.unit.actions.UnitActions.getUnitActions(UnitActions.kt:36)
  at com.unciv.ui.screens.worldscreen.unit.actions.UnitActionsTable.update(UnitActionsTable.kt:28)
  at com.unciv.ui.screens.worldscreen.WorldScreen.update(WorldScreen.kt:366)
  at com.unciv.ui.screens.worldscreen.WorldScreen.render(WorldScreen.kt:720)
  at com.badlogic.gdx.Game.render(Game.java:48)
  at com.unciv.UncivGame.access$render$s2211858(UncivGame.kt:126)
  at com.unciv.UncivGame$wrappedCrashHandlingRender$1.invoke(UncivGame.kt:146)
  at com.unciv.UncivGame$wrappedCrashHandlingRender$1.invoke(UncivGame.kt:146)
  at com.unciv.ui.crashhandling.CrashHandlingExtensionsKt$wrapCrashHandling$1.invoke(CrashHandlingExtensions.kt:17)
  at com.unciv.ui.crashhandling.CrashHandlingExtensionsKt$wrapCrashHandlingUnit$1.invoke(CrashHandlingExtensions.kt:33)
  at com.unciv.ui.crashhandling.CrashHandlingExtensionsKt$wrapCrashHandlingUnit$1.invoke(CrashHandlingExtensions.kt:33)
  at com.unciv.UncivGame.render(UncivGame.kt:459)
  at com.badlogic.gdx.backends.lwjgl3.Lwjgl3Window.update(Lwjgl3Window.java:387)
  at com.badlogic.gdx.backends.lwjgl3.Lwjgl3Application.loop(Lwjgl3Application.java:192)
  at com.badlogic.gdx.backends.lwjgl3.Lwjgl3Application.<init>(Lwjgl3Application.java:166)
  at com.unciv.app.desktop.DesktopLauncher.main(DesktopLauncher.kt:82)
​
```

**Comment by XardasXP** [comment]
The only save I have is from multiplayer game while playing as Arabia. My friend is playing as England and had this bug in Tyre.
[https://upload.disroot.org/r/7uqMyacL#mlNeuumZ4Yv9H8AoCDOBvaKqyIrfd5vGNFwiyvds5cQ=](url)

**Comment by yairm210** [comment]
The root cause of this problem is that the unit was able to be in the city center to begin with, which was caused by not teleporting, because the tile was *passable* even though it was not *enterable*
This is because there is an extra check in canMoveTo that is missing from canPassThrough
Our options were to A. remove the unit from the tile and then check if it can enter, or B replicate the same check for teleportation.

For clarity and to avoid "remove unit from tile and now he has no tile" edge cases I opted to go for B, we can change this in the future if we see there are more things that should be checked only for enter and not for pass through.
